activerecord/lib/active_record/insert_all.rb

```diff
@@ -198,11 +198,17 @@ def updatable_columns
         def touch_model_timestamps_unless(&block)
           model.timestamp_attributes_for_update_in_model.filter_map do |column_name|
             if touch_timestamp_attribute?(column_name)
-              "#{column_name}=(CASE WHEN (#{updatable_columns.map(&block).join(" AND ")}) THEN #{model.quoted_table_name}.#{column_name} ELSE CURRENT_TIMESTAMP END),"
+              "#{column_name}=(CASE WHEN (#{updatable_columns.map(&block).join(" AND ")}) THEN #{model.quoted_table_name}.#{column_name} ELSE #{quoted_current_timestamp_for(column_name)} END),"
             end
           end.join
         end
 
+        def quoted_current_timestamp_for(column_name)
+          connection.quote(
+            model.type_for_attribute(column_name).serialize(model.current_time_from_proper_timezone)
+          )
+        end
```

Comment:
My initial approach tried to dynamically set the precision of `CURRENT_TIMESTAMP` (e.g. `CURRENT_TIMESTAMP(6)` for a `precision: 6` column). This is problematic, as not all databases support extra precision on `CURRENT_TIMESTAMP` (e.g. SQLite does not).

**Is there any reason doing this is a bad idea?**